plugins/pluginManager.js

```diff
@@ -815,14 +816,16 @@ var pluginManager = function pluginManager() {
                 return callback(errors);
             }
             var cwd = eplugin ? eplugin.rfs : path.join(__dirname, plugin);
-            exec('sudo npm install --unsafe-perm', {cwd: cwd}, function(error2) {
-                if (error2) {
-                    errors = true;
-                    console.log('error: %j', error2);
-                }
-                console.log('Done installing plugin %j', plugin);
-                callback(errors);
-            });
+            if (!self.getConfig("api").offline_mode) {
+                exec('sudo npm install --unsafe-perm', {cwd: cwd}, function(error2) {
+                    if (error2) {
+                        errors = true;
+                        console.log('error: %j', error2);
+                    }
+                    console.log('Done installing plugin %j', plugin);
+                    callback(errors);
+                });
+            }
         });
```

Comment:
For this to work properly, we need to enclose this: https://github.com/Countly/countly-server/blob/master/plugins/plugins/api/api.js#L44

into plugins.loadConfigs call